lib/engine/chelsea_bridge.ex

```diff
@@ -2,56 +2,127 @@ defmodule Engine.ChelseaBridge do
   use GenServer
   require Logger
 
+  alias Signs.Utilities.EtsUtils
+
+  # API constants
+  @base_api_url "https://www.chelseabridgesys.com/api/api/"
+  @api_status_endpoint "BridgeRealTime"
+  @api_token_endpoint "token"
+
+  @bridge_table_name :bridge_status
+
+  @type token :: %{
+          value: String.t() | nil,
+          expiration: DateTime.t() | nil
+        }
+
+  @type state :: %{
+          table: :ets.tab(),
+          token: token
+        }
+
   @callback bridge_status() :: %{raised: boolean() | nil, estimate: DateTime.t() | nil}
-  def bridge_status() do
-    case :ets.lookup(:bridge_status, :value) do
+  def bridge_status(ets_table_name \\ @bridge_table_name) do
+    case :ets.lookup(ets_table_name, :value) do
       [{:value, data}] -> data
       _ -> %{raised?: nil, estimate: nil}
     end
   end
 
-  def start_link(_) do
-    GenServer.start_link(__MODULE__, nil, name: __MODULE__)
+  def start_link(opts \\ []) do
+    name = opts[:gen_server_name] || __MODULE__
+    GenServer.start_link(__MODULE__, opts, name: name)
   end
 
-  def init(_) do
+  def init(opts) do
+    bridge_ets_table =
+      :ets.new(opts[:bridge_ets_table_name] || @bridge_table_name, [
+        :named_table,
+        read_concurrency: true
+      ])
+
+    state = %{
+      table: bridge_ets_table,
+      token: %{value: nil, expiration: nil}
+    }
+
     send(self(), :update)
-    :ets.new(:bridge_status, [:named_table, read_concurrency: true])
-    {:ok, %{}}
+
+    {:ok, state}
   end
 
   def handle_info(:update, state) do
     Process.send_after(self(), :update, 60_000)
-    url = Application.get_env(:realtime_signs, :chelsea_bridge_url)
-    auth = Application.get_env(:realtime_signs, :chelsea_bridge_auth)
     http_client = Application.get_env(:realtime_signs, :http_client)
+    now = Timex.now()
+
+    token =
+      if(state.token.expiration == nil or DateTime.compare(state.token.expiration, now) == :lt) do
```

Comment:
nit: `Timex` has some slightly more ergonomic functions for comparisons, like `Timex.before?` and `Timex.after?`

Reply:
Thanks for pointing out! Switched over to use `Timex.before` here, and `Timex.shift` elsewhere, which I now prefer to the DateTime equivalent.